applications/mne_analyze/libs/anShared/Model/ecdsetmodel.cpp

```diff
@@ -0,0 +1,195 @@
+// INCLUDES
+//=============================================================================================================
+
+#include "ecdsetmodel.h"
+#include "../Utils/types.h"
+#include "../Utils/metatypes.h"
+#include <inverse/dipoleFit/ecd_set.h>
+#include <inverse/dipoleFit/dipole_fit.h>
+#include <inverse/dipoleFit/dipole_fit_settings.h>
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// Eigen INCLUDES
+//=============================================================================================================
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace ANSHAREDLIB;
+using namespace INVERSELIB;
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DEFINE GLOBAL METHODS
+//=============================================================================================================
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+EcdSetModel::EcdSetModel(DipoleFitSettings *pDipolSettings, QObject *pParent)
+    :AbstractModel(pParent)
+{
+    DipoleFit dipFit(pDipolSettings);
+    m_ecdSet = dipFit.calculateFit();
+}
+
+
+//*************************************************************************************************************
+
+EcdSetModel::EcdSetModel(const QString &sDipFileName, QObject *pParent)
+{
+    m_ecdSet = ECDSet::read_dipoles_dip(sDipFileName);
+}
+
+
+//*************************************************************************************************************
+
+QVariant EcdSetModel::data(const QModelIndex &index, int role) const
+{
+    if(!index.isValid()) {
+        return QVariant();
```

Comment:
More concise notation for early return case would be possible (as used in setData).